lib/index.js

```diff
@@ -140,6 +140,14 @@ function getOptions(options) {
   options.sourceMap = getSourceMap(options);
   options.style = getStyle(options) || 0;
 
+  // context object represents node-sass environment
+  options.context = {
+    info: module.exports.info,
+    state: {},
+    sync: false,
```

Comment:
Totally makes sense. Thanks for explaining the logic and rationale. :)

For 1) if we don't restrict writing to the options hash, technically any new options we add risk breaking backwards compatibility. We could just tell import / custom function devs to namespace their items stored in the options hash to avoid conflicts.

For 2) importers used in gulp or grunt pipelines use the same options collection repeatedly. It seemed safer to work on a clone of the options. Also happy to pull this out.

More than happy to remove any/all of these items if we think they're outside the goals of node-sass.